contracts/protocol/pool/PoolApeStaking.sol

```diff
@@ -489,6 +492,105 @@ contract PoolApeStaking is
+        address tmpPositionOwner;
+    }
+
+    /// @inheritdoc IPoolApeStaking
+    function claimPairedApeRewardAndCompound(
+        address nftAsset,
+        address[] calldata users,
+        ApeCoinStaking.PairNft[][] calldata _nftPairs
+    ) external nonReentrant {
+        require(users.length == _nftPairs.length, "invalid parameter");
+        DataTypes.PoolStorage storage ps = poolStorage();
+
+        CompoundPairedApeRewardLocalVar memory localVar;
+        localVar.xTokenAddress = ps._reserves[nftAsset].xTokenAddress;
+        localVar.bakcContract = INTokenApeStaking(localVar.xTokenAddress)
+            .getBAKC();
+        localVar.bakcNToken = ps
+            ._reserves[address(localVar.bakcContract)]
+            .xTokenAddress;
+        localVar.balanceBefore = APE_COIN.balanceOf(address(this));
+        localVar.amounts = new uint256[](_nftPairs.length);
+
+        for (uint256 i = 0; i < _nftPairs.length; i++) {
+            ApeCoinStaking.PairNft[] calldata nftPair = _nftPairs[i];
+            localVar.originArray = new address[](nftPair.length);
+            for (uint256 j = 0; j < nftPair.length; j++) {
+                localVar.tmpPositionOwner = INToken(localVar.xTokenAddress)
+                    .ownerOf(nftPair[j].mainTokenId);
+                require(
+                    users[i] == localVar.tmpPositionOwner,
+                    "user is not owner"
+                );
+
+                localVar.originArray[j] = validateBAKCOwnerAndTransfer(
+                    localVar.bakcContract,
+                    nftPair[j].bakcTokenId,
+                    localVar.bakcNToken,
+                    localVar.xTokenAddress,
+                    users[i]
+                );
+            }
+
+            INTokenApeStaking(localVar.xTokenAddress).claimBAKC(
+                nftPair,
+                address(this)
+            );
+
+            for (uint256 index = 0; index < nftPair.length; index++) {
+                localVar.bakcContract.safeTransferFrom(
+                    localVar.xTokenAddress,
+                    localVar.originArray[index],
+                    nftPair[index].bakcTokenId
+                );
+            }
+
+            localVar.balanceAfter = APE_COIN.balanceOf(address(this));
+            localVar.amounts[i] =
+                localVar.balanceAfter -
+                localVar.balanceBefore;
+            localVar.balanceBefore = localVar.balanceAfter;
+            localVar.totalAmount += localVar.amounts[i];
+        }
+
+        uint256 compoundFee = ps._apeCompoundFee;
+        uint256 totalFee = localVar.totalAmount.percentMul(compoundFee);
+        APE_COMPOUND.deposit(address(this), localVar.totalAmount);
+
+        if (totalFee > 0) {
+            IERC20(address(APE_COMPOUND)).safeTransfer(msg.sender, totalFee);
+        }
+
+        for (uint256 index = 0; index < users.length; index++) {
+            if (localVar.amounts[index] != 0) {
+                _supplyCApeForUser(
```

Comment:
can we have common function defined for claimApeAndCompound and this function? seems too many duplicated code

Reply:
`_supplyCApeForUser` may fail due to the precision loss of:

```
APE_COMPOUND.deposit(address(this), localVar.totalAmount);
```

Reply:
@GopherJ  how would precision loss occur here?

Reply:
@GopherJ  Do you have any data case. Precision loss is true, but I don't think it will cause _supplyCApeForUser fail.